Lib/importlib/resources.py

```diff
@@ -238,15 +238,15 @@ def contents(package: Package) -> Iterator[str]:
     package = _get_package(package)
     reader = _get_resource_reader(package)
     if reader is not None:
-        yield from reader.contents()
-        return
+        return reader.contents()
     # Is the package a namespace package?  By definition, namespace packages
     # cannot have resources.  We could use _check_location() and catch the
     # exception, but that's extra work, so just inline the check.
-    if package.__spec__.origin is None or not package.__spec__.has_location:
+    elif package.__spec__.origin is None or not package.__spec__.has_location:
         return []
-    package_directory = Path(package.__spec__.origin).parent
-    yield from os.listdir(str(package_directory))
+    else:
+        package_directory = Path(package.__spec__.origin).parent
+        return os.listdir(str(package_directory))
```

Comment:
`str()` is not needed here. `os.listdir()` supports the path protocol.

`str()` can be removed also in `_normalize_path()`. Actually calling `str()` in `_normalize_path()` can mask an error when wrong type is passed.